src/expr/src/vector_op/arithmetic_op.rs

```diff
@@ -108,6 +108,21 @@ pub fn decimal_abs(decimal: Decimal) -> Result<Decimal> {
     Ok(Decimal::abs(&decimal))
 }
 
+#[inline(always)]
+pub fn general_pow<T1, T2, T3>(l: T1, r: T2) -> Result<T3>
```

Comment:
If we cast them in frontend, I guess we can only support a concrete implementation (f64,f64)->f64 instead of a general one.

Reply:
At least the `Decimal` version needs to be implemented besides the f64 version.
 
<img width="1056" alt="image" src="https://user-images.githubusercontent.com/10192522/217729222-b5057a05-e559-4f38-be8a-2d38efe6a5e2.png">